src/pages/iv-items/forms/ItemsForm.js

```diff
@@ -0,0 +1,575 @@
+  const { platformLabels } = useContext(ControlContext)
+  const [showLotCategories, setShowLotCategories] = useState(false)
+  const [showSerialProfiles, setShowSerialProfiles] = useState(false)
+  const { recordId } = store
+
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const invalidate = useInvalidate({
+    endpointId: InventoryRepository.Items.qry
+  })
+  const imageUploadRef = useRef(null)
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: null,
+      name: '',
+      sku: '',
+      categoryId: null,
+      flName: '',
+      shortName: '',
+      groupId: null,
+      msId: null,
+      description: '',
+      priceType: null,
+      ptName: '',
+      procurementMethod: null,
+      valuationMethod: null,
+      volume: '',
+      weght: '',
+      trackBy: null,
+      unitPrice: '',
+      ivtItem: true,
+      salesItem: true,
+      purchaseItem: true,
+      kitItem: false,
+      taxId: null,
+      lotCategoryId: null,
+      spfId: '',
+      categoryName: '',
+      defSaleMUId: '',
+      pgId: ''
+    },
+    access,
+    enableReinitialize: true,
+    validateOnChange: true,
+
+    validationSchema: yup.object({
+      categoryId: yup.string().required(),
+      sku: yup.string().required(),
+      name: yup.string().required(),
+      priceType: yup.string().required(),
+      msId: yup.string().required(),
+      lotCategoryId: yup
+        .string()
+        .nullable()
+        .test(function (value) {
+          const { trackBy } = this.parent
+
+          return trackBy === '2' || trackBy === 2 ? value != null && value.trim() !== '' : true
+        }),
+      spfId: yup
+        .string()
+        .nullable()
+        .test(function (value) {
+          const { trackBy } = this.parent
+
+          return trackBy === '1' || trackBy === 1 ? value != null && value.trim() !== '' : true
+        })
+    }),
+    onSubmit: async obj => {
+      const recordId = obj.recordId
+
+      try {
+        const response = await postRequest({
+          extension: InventoryRepository.Items.set,
+          record: JSON.stringify({ ...obj, attachment: null })
+        })
+
+        if (!recordId) {
+          toast.success(platformLabels.Added)
+          formik.setValues({
+            ...obj,
+            recordId: response.recordId
+          })
+          setStore(prevStore => ({
+            ...prevStore,
+            recordId: response.recordId,
+            _msId: formik.values.msId
+          }))
+        } else {
+          toast.success(platformLabels.Edited)
+        }
+        setFormikInitial(formik.values)
+
+        if (imageUploadRef.current) {
+          await imageUploadRef.current.submit()
+        }
+
+        const res = await getRequest({
+          extension: InventoryRepository.Items.get,
+          parameters: `_recordId=${response.recordId}`
+        })
+
+        formik.setFieldValue('sku', res.record.sku)
+      } catch (error) {}
```

Comment:
try removing try catch and cause an api error, if the error window shows and nothing break leave them removed